src/ext/serializer.c

```diff
@@ -16,6 +16,8 @@
 
 ZEND_EXTERN_MODULE_GLOBALS(ddtrace);
 
+#define DD_TRACE_MAX_PID_LEN 7  // Theoretical maximum pid on 64-bit: 2^22 (4194304)
```

Comment:
Where does the limit of `22` come from?

Reply:
This is from the [man pages](http://man7.org/linux/man-pages/man5/proc.5.html).

> On 64-bit systems, pid_max can be set to any value up to 2^22 (PID_MAX_LIMIT, approximately 4 million).

Reply:
On Linux you can change the max PIDs at runtime -- I would be safe and use the max for the PHP long. There is a define for it in PHP already: `MAX_LENGTH_OF_LONG`, which is `20` for 64 bit.

Reply:
Thanks for the suggestion! Updated.